### DISCUSSION TITLE
Turn Stage Panel Widget On/Off

### DISCUSSION THREAD

**Discussion by DeepakMalkan** [iTwin/itwinjs-core#1629] [Q&A] [answered]
I have a widget add to the Right Stage Panel. Under certain conditions, I'd like to turn it OFF, and then under certain other conditions, I'd like to turn it ON.

How can I do this?

I tried to set stateFunc prop of the widget, but it doesn't seem to work.

**Comment by NancyMcCallB** [comment, marked as answer]
Hi, Deepak,

Could you clarify some things for us? 
1. Are you trying to turn off the panel or an individual widget?
2. Which version of AppUi are you using? (1 or 2)?

@GerardasB any thoughts?

**Reply by DeepakMalkan** [reply]
Hi Nancy,

I'm trying to turn off just the individual widget that I've created/registered with the Panel.

I'm not sure which AppUi version I'm using. I created my App using 'create-react-app' .. hopefully this helps.

I can give a quick demo of what I'd like to achieve.

Deepak.

**Reply by bsteinbk** [reply]
I believe the latest CRA creates an app that runs in AppUi 2. If that is the case you can hide or show the entire right panel using code like the following.
```
const frontstageDef = FrontstageManager.activeFrontstageDef;
const panelDef = frontstageDef?.getStagePanelDef(location);  // location = StagePanelLocation.Right
panelDef.panelState = panelState; //panelState = StagePanelState.Minimized or StagePanelState.Open
```
If you just want to hide of show a single widget's tab, then you use code like the following.
```
const frontstageDef = FrontstageManager.activeFrontstageDef;
const widgetDef = frontstageDef?.findWidgetDef(widgetId); // widgetId = unique Id you assigned your widget
widgetDef?.setWidgetState(s);  // s = WidgetState.Open or WidgetState.Closed
```

**Reply by DeepakMalkan** [reply]
Thanks Bill.

This works and helps, but not exactly what I had in mind. With WidgetState.Closed, the widget is still there and accessible to the user. Is there a way to completely hide the widget so that the user cannot access it?

Deepak.

**Reply by bsteinbk** [reply]
Try WidgetState.Hidden instead of WidgetState.Closed.

**Reply by DeepakMalkan** [reply]
Ahh.. I should have checked myself.

Thx. Its works the way I want it now!
